### PR TITLE
[feature] add parameter to generate_code that specifies whether to pr…

### PR DESCRIPTION
**Pull request by poshul**
…ecede foreign imports with a .

Fixes issues with OpenMS compilation  and language_level=3.

### CODE REVIEW

**Comment by poshul** [comment]
Since we're discussing this in OpenMS/OpenMS#4922 I'm reopening this,

File: tests/test_full_library.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
 from __future__ import absolute_import, print_function
 
 import os
+<<<<<<< HEAD
+=======
+import sys
+import glob
+
+import pytest
+>>>>>>> jpfeuffer/feature/relative_import
```
Comment (jpfeuffer) [inline tests/test_full_library.py:10]:
```suggestion
import glob
```

Comment (jpfeuffer) [reply tests/test_full_library.py:10]: Should be the only thing we need.

**Comment by jpfeuffer** [comment]
@uweschmitt Seems like your Travis does not work correctly anymore since some time. It does not report back the build results anymore. Would be great if you could look into [that](https://travis-ci.community/t/github-pr-is-being-built-but-result-is-not-shown/7025).

**Comment by jpfeuffer** [comment]
Regarding the PR. We changed the test to be a real package and now it correctly reproduces the errors that we are encountering. Relative paths are needed to correctly import other pxds from the same package in python 3.
This is the first solution by introducing a flag. We plan to introduce something like a package_root parameter though to also support deeper hierarchies.

**Comment by uweschmitt** [comment]
> @uweschmitt Seems like your Travis does not work correctly anymore since some time. It does not report back the build results anymore. Would be great if you could look into [that](https://travis-ci.community/t/github-pr-is-being-built-but-result-is-not-shown/7025).

What I see is that the test suite fails because of some changes in the tests which are not present in the master branch. Please check the output at https://travis-ci.org/github/uweschmitt/autowrap/jobs/728786298

Do your tests succeed locally?

**Comment by uweschmitt** [comment]
Looks like

removal of the line
https://github.com/poshul/autowrap/commit/9a84fb7689daebf7d2aaf5d60c665295723ccc05#diff-8210f5faf5a0c606f2581716531303ecL180

breaks the tests.

File: tests/test_full_library.py [hunk 2]
```diff
@@ -169,9 +170,10 @@ def test_full_lib(tmpdir):
     are not an issue.
     """
 
-    curdir = os.getcwd()
-
-    os.chdir(tmpdir.strpath)
+    workdir = tmpdir.strpath + "/package"
```
Comment (jpfeuffer) [inline tests/test_full_library.py:173]:
```suggestion
    curdir = os.getcwd()
    workdir = tmpdir.strpath + "/package"
```

**Comment by jpfeuffer** [comment]
You are correct, thank you. This seems to be the culprit. Probably an error during merge resolution. Yes, it works in a similar version on H.Roest's branch here: https://github.com/hroest/autowrap/pull/2
Should be minor things now.

File: tests/test_full_library.py [hunk 3]
```diff
@@ -234,22 +237,21 @@ def test_full_lib(tmpdir):
 
         # Step 4: Generate CPP code
```
Comment (jpfeuffer) [inline tests/test_full_library.py:238]:
```suggestion
        os.chdir("..")
        # Step 4: Generate CPP code
```

Comment (jpfeuffer) [reply tests/test_full_library.py:238]: Might also work if you change the relative filenames in the following steps and remove "package", but that is how I did it on my branch.

Comment (jpfeuffer) [reply tests/test_full_library.py:238]: @poshul Any updates on this? If you just accept the change, I can check in travis if it was enough.

File: tests/test_full_library.py [hunk 4]
```diff
@@ -231,25 +234,24 @@ def test_full_lib(tmpdir):
                 allDecl=masterDict,
             )
```
Comment (jpfeuffer) [inline tests/test_full_library.py:235]:
```suggestion
                allDecl=masterDict,
                add_relative=True
            )
```

Now we hopefully just need to specify that this is a cython package and activate your new flag.

**Comment by jpfeuffer** [comment]
👍🏻  works.

@uweschmitt What do you think about this approach for a quick fix for python3 projects whose pxds are inside a package?

In the long term we envision extending this to automatically infer if files belong to a package and also to support deeper hierarchies by parsing full relative paths.

**Comment by jpfeuffer** [comment]
@uweschmitt @hroest How do we proceed?

**Comment by uweschmitt** [comment]
Test `tests/test_full_library.py` fails after merge.

**Comment by jpfeuffer** [comment]
Ok, what is the difference between the two test environments?

**Comment by jpfeuffer** [comment]
https://travis-ci.org/github/uweschmitt/autowrap/builds/759469830

Looks fine to me. Did you execute the tests locally somewhere? What was the exact error?

**Comment by uweschmitt** [comment]
I made a mistake. I merged it now without problems and will upload the update to pypi.

**Pull request merged**